core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -739,56 +738,57 @@ protected File getRootDirFor(String name) {
 
     /**
      * Gets the {@link Jenkins} singleton.
-     * {@link #getInstanceOrNull()} provides the unchecked versions of the method.
      * @return {@link Jenkins} instance
-     * @throws IllegalStateException {@link Jenkins} has not been started, or was already shut down
-     * @since 1.590
-     * @deprecated use {@link #getInstance()}
+     * @throws IllegalStateException for the reasons that {@link #getInstanceOrNull} might return null
+     * @since FIXME
      */
-    @Deprecated
     @Nonnull
-    public static Jenkins getActiveInstance() throws IllegalStateException {
-        Jenkins instance = HOLDER.getInstance();
+    public static Jenkins get() throws IllegalStateException {
+        Jenkins instance = getInstanceOrNull();
         if (instance == null) {
-            throw new IllegalStateException("Jenkins has not been started, or was already shut down");
+            throw new IllegalStateException("Jenkins.instance is missing. Read the documentation of Jenkins.getInstanceOrNull to see what you are doing wrong.");
```

Comment:
if Jenkins.instance where missing you would blow up with some form of linkage error ->  the instance is not missing it is `null`

Reply:
I am not sure what the distinction is. If you got this far, obviously `Jenkins.class` has been successfully loaded, and `theInstance == null`.

Reply:
The distinction is massive in terms of clarity

Jenkins.Instance is a field. 
A missing field is saying the field is not there which can only be because it was removed from a class (it is no longer defined in the source file yet you expect it to be there) 
Null is not missing, the field is there it has just not been set to anything other than its default (null) (or has been set back to null for some reason) 

The message should help users (is this something wrong with their job, jenkins...) 
Basically the message should be Jenkins.instance is null (unset)   This is a bug in the plugin. Please read the docume....